datafusion/optimizer/tests/integration-test.rs

```diff
@@ -46,6 +52,17 @@ fn case_when() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn case_when_aggregate() -> Result<()> {
+    let sql = "SELECT col_utf8, SUM(CASE WHEN col_int32 > 0 THEN 1 ELSE 0 END) AS n FROM test GROUP BY col_utf8";
+    let plan = test_sql(sql)?;
+    let expected = "Projection: #test.col_utf8, #SUM(CASE WHEN test.col_int32 > Int64(0) THEN Int64(1) ELSE Int64(0) END) AS n\
```

Comment:
Could you merge latest from master - we should not include `#` before column names now

Reply:
I have fixed the issue -- it was rewriting sorts. New changes very close

Reply:
Will do

Reply:
in e3830a7bd